backend/src/main/java/ai/verta/modeldb/advancedService/AdvancedServiceImpl.java

```diff
@@ -963,59 +951,9 @@ public void findHydratedExperiments(
             ModelDBServiceActions.READ);
       }
 
-      ExperimentServiceImpl experimentService =
-          new ExperimentServiceImpl(
-              authService, roleService, experimentDAO, projectDAO, artifactStoreDAO);
-      if (!request.getExperimentIdsList().isEmpty()) {
-        List<String> accessibleExperimentIds =
-            experimentService.getAccessibleExperimentIDs(
-                request.getExperimentIdsList(), ModelDBServiceActions.READ);
-        if (accessibleExperimentIds.isEmpty()) {
-          String errorMessage =
-              "Access is denied. User is unauthorized for given Experiment IDs : "
-                  + accessibleExperimentIds;
-          ModelDBUtils.logAndThrowError(
-              errorMessage,
-              Code.PERMISSION_DENIED_VALUE,
-              Any.pack(FindExperiments.getDefaultInstance()));
-        }
-        request =
-            request
-                .toBuilder()
-                .clearExperimentIds()
-                .addAllExperimentIds(accessibleExperimentIds)
-                .build();
-      }
-
-      for (KeyValueQuery predicate : request.getPredicatesList()) {
-        // ID predicates only supported for EQ
-        if (predicate.getKey().equals(ModelDBConstants.ID)) {
-          if (!predicate.getOperator().equals(OperatorEnum.Operator.EQ)) {
-            Status statusMessage =
-                Status.newBuilder()
-                    .setCode(Code.INVALID_ARGUMENT_VALUE)
-                    .setMessage(ModelDBConstants.NON_EQ_ID_PRED_ERROR_MESSAGE)
-                    .build();
-            throw StatusProto.toStatusRuntimeException(statusMessage);
-          }
-          List<String> accessibleExperimentIds =
-              experimentService.getAccessibleExperimentIDs(
-                  Collections.singletonList(predicate.getValue().getStringValue()),
-                  ModelDBServiceActions.READ);
-
-          if (accessibleExperimentIds.isEmpty()) {
-            String errorMessage =
-                "Access is denied. User is unauthorized for given Experiment IDs : "
-                    + accessibleExperimentIds;
-            ModelDBUtils.logAndThrowError(
-                errorMessage,
-                Code.PERMISSION_DENIED_VALUE,
-                Any.pack(FindExperiments.getDefaultInstance()));
-          }
-        }
-      }
-
```

Comment:
Move this validation into the FindExperiment DAO Impl for the common code purpose